frontend/src/component/PetGallery.tsx

```diff
@@ -8,13 +8,13 @@ import useAuth from "../hooks/useAuth";
 
 type PetGalleryProps = {
     pets: Pet[]
-
     navigateTo: string | undefined
 }
 
 export default function PetGallery(props: PetGalleryProps) {
     const user = useAuth(true)
-    const pets = props.pets.map((pet: Pet) => {
+    const filteredPets = props.pets.filter(pet => pet.userId === user?.id);
```

Comment:
I suggest filtering the pets in the backend and only returning the user's own pets to the frontend